tests/infrastructure/conftest.py

```diff
@@ -0,0 +1,26 @@
+import bitmath
+import pytest
+
+
+@pytest.fixture(scope="session")
+def hugepages_gib_values(workers):
+    """Return the list of hugepage sizes (in GiB) across all worker nodes."""
+    return [
+        int(bitmath.parse_string_unsafe(value).GiB)
+        for worker in workers
+        if (value := worker.instance.status.allocatable.get("hugepages-1Gi"))
+    ]
+
+
+@pytest.fixture(scope="session")
+def xfail_if_no_huge_pages(hugepages_gib_values):
+    """Mark tests as xfail if the cluster lacks 1Gi hugepages."""
+    if not hugepages_gib_values or max(hugepages_gib_values) < 1:
+        pytest.xfail("Requires at least 1Gi hugepages on some node")
+
+
+@pytest.fixture(scope="session")
+@pytest.mark.usefixtures("xfail_if_no_huge_pages")
+def hugepages_gib_max(hugepages_gib_values):
```

Comment:
```suggestion
def hugepages_gib_max(hugepages_gib_values, xfail_if_no_huge_pages):
```

Reply:
`xfail_if_no_huge_pages` depends on `hugepages_gib_values`, so logically it should be called after

Also, I see you also still call `xfail_if_no_huge_pages` in old place:
https://github.com/RedHatQE/openshift-virtualization-tests/pull/2381/files#diff-f4f868da16a297cb9b2d2528aa94fed805889a495a4536e1eff09f028cd6f525R24

no point for having it here and there.. 

and one more - for hugepages we have a pytest mark `hugepages`, so probably you can add the sanity check - fail at the beginning of the test run if test mark "hugepages" selected, but the cluster does not have it.

Reply:
https://issues.redhat.com/browse/CNV-71286 , with this in i can use markers and prevent calling these checks.

Reply:
you can add marker now - it will help you to filter out all hugepages tests when they not needed

Reply:
added markers